### PR TITLE
Add replicate/optimize Current Environment to IGCE data collection script

### PR DESCRIPTION
**Pull request by zachclark-ccpo**
# Overview
- Added support for Current Environment mapping
- Added validation for IGCE Estimate

## Testing
`ACQ0000480` output should be:
```
{
    "documentType": "INDEPENDENT_GOVERNMENT_COST_ESTIMATE",
    "templatePayload": {
        "fundingDocument": {
            "fundingType": "MIPR",
            "miprNumber": ""
        },
        "periodsEstimate": [
            {
                "period": {
                    "periodType": "BASE",
                    "periodUnitCount": "1",
                    "periodUnit": "YEAR",
                    "optionOrder": "1"
                },
                "periodLineItems": [
                    {
                        "idiqClin": "x002/x018 Cloud Support UNCLASSIFIED",
                        "contractType": "TBD",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Training #1",
                        "itemDescription": "Need and duration",
                        "unitPrice": 20,
                        "quantity": 10,
                        "unit": "months"
                    },
                    {
                        "idiqClin": "x007 Travel",
                        "contractType": "T&M",
                        "dowTaskNumber": "10f",
                        "serviceTitle": "Travel",
                        "itemDescription": "3 x (Dagobah System, 20 days, 1 travelers); ",
                        "unitPrice": 20,
                        "quantity": 1,
                        "unit": "period"
                    },
                    {
                        "idiqClin": "x003/x019 Cloud SECRET CLASSIFIED",
                        "contractType": "FFP",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Objective-based requirements",
                        "itemDescription": "Optimize current environment (2 instances)",
                        "unitPrice": "700",
                        "quantity": 1,
                        "unit": "period"
                    },
                    {
                        "idiqClin": "x001/x017 Cloud UNCLASSIFIED",
                        "contractType": "FFP",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Developer Tools - Cyber Tools",
                        "itemDescription": "blah blah blah IL2",
                        "unitPrice": 78,
                        "quantity": 12,
                        "unit": "month"
                    },
                    {
                        "idiqClin": "x001/x017 Cloud UNCLASSIFIED",
                        "contractType": "FFP",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Compute - Instance #1",
                        "itemDescription": "25 x (COOP_DISASTER_RECOVERY, serverless, NEW, 6 vCPUs, 6 GB RAM, object storage: 6 TB, MEMORY)",
                        "unitPrice": 700,
                        "quantity": 12,
                        "unit": "month"
                    }
                ]
            },
            {
                "period": {
                    "periodType": "OPTION",
                    "periodUnitCount": "36",
                    "periodUnit": "WEEK",
                    "optionOrder": "2"
                },
                "periodLineItems": [
                    {
                        "idiqClin": "x002/x018 Cloud Support UNCLASSIFIED",
                        "contractType": "TBD",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Training #1",
                        "itemDescription": "Need and duration",
                        "unitPrice": 20,
                        "quantity": 3,
                        "unit": "months"
                    },
                    {
                        "idiqClin": "x003/x019 Cloud SECRET CLASSIFIED",
                        "contractType": "FFP",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Objective-based requirements",
                        "itemDescription": "Optimize current environment (2 instances)",
                        "unitPrice": "800",
                        "quantity": 1,
                        "unit": "period"
                    },
                    {
                        "idiqClin": "x001/x017 Cloud UNCLASSIFIED",
                        "contractType": "FFP",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Compute - Instance #1",
                        "itemDescription": "25 x (COOP_DISASTER_RECOVERY, serverless, NEW, 6 vCPUs, 6 GB RAM, object storage: 6 TB, MEMORY)",
                        "unitPrice": 700,
                        "quantity": 9,
                        "unit": "month"
                    }
                ]
            },
            {
                "period": {
                    "periodType": "OPTION",
                    "periodUnitCount": "252",
                    "periodUnit": "DAY",
                    "optionOrder": "3"
                },
                "periodLineItems": [
                    {
                        "idiqClin": "x002/x018 Cloud Support UNCLASSIFIED",
                        "contractType": "TBD",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Training #1",
                        "itemDescription": "Need and duration",
                        "unitPrice": 20,
                        "quantity": 4,
                        "unit": "months"
                    },
                    {
                        "idiqClin": "x003/x019 Cloud SECRET CLASSIFIED",
                        "contractType": "FFP",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Objective-based requirements",
                        "itemDescription": "Optimize current environment (2 instances)",
                        "unitPrice": "900",
                        "quantity": 1,
                        "unit": "period"
                    },
                    {
                        "idiqClin": "x001/x017 Cloud UNCLASSIFIED",
                        "contractType": "FFP",
                        "dowTaskNumber": "TBD",
                        "serviceTitle": "Developer Tools - Data Management",
                        "itemDescription": "blah blah blah IL4",
                        "unitPrice": 20,
                        "quantity": 9,
                        "unit": "month"
                    }
                ]
            }
        ],
        "contractingShop": {
            "name": "DITCO"
        },
        "instructions": {
            "estimateDescription": "Figured it out",
            "assumptionsMade": "assumptions",
            "toolsUsed": "AWS pricing calculator,Oracle Cloud pricing calculator,Prices previously paid on a same or similar contract,Epic Software That I Created",
            "informationSource": "info source",
            "previousEstimateComparison": "The previous cost estimate was 4% less than the actual prices paid."
        },
        "surgeCapabilities": 20
    }
}
```

Make sure current environment will map with the correct inputs.

### CODE REVIEW

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml [hunk 1]
```diff
@@ -163,61 +172,76 @@ Igce.prototype = {
...
+            && item_to_validate.classification_level
+            && item_to_validate.idiq_clin_type
+            && item_to_validate.contract_type
+            && item_to_validate.description) {
+            return true;
+        }
+        else {
+            return false;
+        }
+    },
```
Comment (zachclark-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml:187]: Validate IGCE Estimate script

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml [hunk 2]
```diff
@@ -699,6 +723,51 @@ Igce.prototype = {
...
+                });
+            }
+        }
+        catch (error) {
+            throw this.errUtil.createError(
+                "Igce --> mapCurrentEnvironmentToPeriodsEstimate(): " + error,
+                this.errUtil.METHOD_ERROR
+            );
+        }
+    },
```
Comment (zachclark-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml:769]: mapCurrenEnvironmentToPeriodsEstimate script

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml [hunk 3]
```diff
@@ -886,6 +956,90 @@ Igce.prototype = {
...
+                "contract_type": acq_package.contract_type.getDisplayValue(),
+                "classification_level": highestClassificationLevel,
+                "description": currentEnvOption,
+                "value_option": valueOption,
+                "values": values
+            }
+            this.mapCurrentEnvironmentToPeriodsEstimate(envObject, periodsEstimate);
+
+        }
+
```
Comment (zachclark-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml:1042]: retrieval logic

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml [hunk 4]
```diff
@@ -163,61 +172,76 @@ Igce.prototype = {
...
-                        }
-                        else {
-                            formattedQuantity = 0;
-                        }
-
-
-                        // Insert into periods_estimate, under the specified period and period_line_items 
-                        periods_estimate[indexOfCurrentPeriod].periodLineItems.push({
+                // If the item meets the validation criteria
+                if (Igce.prototype.validateIgceEstimateItem(item)) {
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml:194]: Nice defensive measure

**Review by jasonburkert-ccpo — APPROVED** [review]

**Comment by jasonburkert-ccpo** [comment]
This resolves bug 2 documented in ticket AT-8737

**Pull request merged**